### ISSUE TITLE
Crash if cover letter contains non-ascii characters

### ISSUE THREAD

**Issue by ehabkost** [stefanha/git-publish#95]
Reproducer can be found at: https://github.com/ehabkost/git-publish/commit/7a7c95f95319d925a75d2790e551c3564a61238f

If the cover letter has non-ascii characters, git-publish crashes with:

```
Traceback (most recent call last):
  File "/tmp/tmp.5mwl8IOgKs/bin/git-publish", line 905, in <module>
    sys.exit(main())
  File "/tmp/tmp.5mwl8IOgKs/bin/git-publish", line 851, in main
    open(cover_letter_path, 'wb').write(msg.as_bytes())
  File "/usr/lib64/python3.8/email/message.py", line 178, in as_bytes
    g.flatten(self, unixfrom=unixfrom)
  File "/usr/lib64/python3.8/email/generator.py", line 116, in flatten
    self._write(msg)
  File "/usr/lib64/python3.8/email/generator.py", line 181, in _write
    self._dispatch(msg)
  File "/usr/lib64/python3.8/email/generator.py", line 214, in _dispatch
    meth(msg)
  File "/usr/lib64/python3.8/email/generator.py", line 432, in _handle_text
    super(BytesGenerator,self)._handle_text(msg)
  File "/usr/lib64/python3.8/email/generator.py", line 249, in _handle_text
    self._write_lines(payload)
  File "/usr/lib64/python3.8/email/generator.py", line 155, in _write_lines
    self.write(line)
  File "/usr/lib64/python3.8/email/generator.py", line 406, in write
    self._fp.write(s.encode('ascii', 'surrogateescape'))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 18-19: ordinal not in range(128)
```

Bisected to commit 0f2dd0bd9e1165a46be733158dbec4c2deb3d8e0.

**Comment by stefanha** [comment]
@stefano-garzarella

**Comment by stefano-garzarella** [comment]
@stefanha I'll try to find an alternative, otherwise you can revert it. 😔

Maybe I need to use `get_content` and then decode it, or avoid to use the `email` module as in the first version and fix the long tag issue.

@ehabkost thanks for reporting this 😉

**Comment by philmd** [comment]
Confirmed:
`UnicodeEncodeError: 'ascii' codec can't encode character '\xe9' in position 21: ordinal not in range(128)`

**Comment by stefanha** [comment]
@philmd: @stefano-garzarella is working on a fix here: #99
